meta-mbl-distro/classes/mbl-partitions.bbclass

```diff
@@ -24,138 +24,407 @@ inherit mbl-artifact-names
 # will be used in your case.
 MBL_FLASH_ERASE_BLOCK_SIZE_KiB ?= "16384"
 
+MBL_DEFAULT_ALIGN_KiB ?= "${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}"
+MBL_DEFAULT_FSTYPE ?= "ext4"
+MBL_DEFAULT_MOUNT_OPTS ?= "defaults"
+MBL_DEFAULT_FS_FREQ ?= "0"
+MBL_DEFAULT_FS_PASSNO ?= "0"
 
+# ------------------------------------------------------------------------------
+# Partition variables
+# ------------------------------------------------------------------------------
 # We have two different flavours of "partition" in MBL:
 # * Normal file system partitions, recorded in the partition table.
 # * Raw non-file system "partitions", not recorded in the partition table.
+# 
+# List of all partitions, in order
+MBL_PARTITION_NAMES = ""
+#
+# For each of these "partitions" there are the following variables:
+# * MBL_<partition>_ALIGN_KiB   - minimum alignment in KiB
+# * MBL_<partition>_SIZE_KiB    - size in KiB
+# * MBL_<partition>_SIZE_MiB    - size in MiB
+# * MBL_<partition>_OFFSET1_KiB - offset of first bank in KiB
+# * MBL_SKIP_<partition>        - skip this partition?
+# * MBL_<partition>_NO_FS       - is this a raw partition (not a file system partition)?
+# * MBL_<partition>_IS_BANKED   - are there two banks of this partition
+#
+# For partitions with two banks there is:
+# * MBL_<partition>_OFFSET2_KiB - offset of second bank in KiB
+#
+# For non-file system partitions there is:
+# * MBL_<partition>_FILENAME    - file in image deploy dir
+#
+# For file system partitions there is:
+# * MBL_<partition>_MOUNT_POINT - where this partition is mounted in the Linux file system)
+# * MBL_<partition>_LABEL       - label of partition
+# * MBL_<partition>_FSTYPE      - type of file system
+# * MBL_<partition>_MOUNT_OPTS  - mount options
+# * MBL_<partition>_FS_FREQ     - fifth fstab(5) field
+# * MBL_<partition>_FS_PASSNO   - sixth fstab(5) field
+#
+# Notes:
+# * MBL_<partition>_ALIGN_KiB may be left unset outside of this file and a
+#   default value will be used. For WKS_BOOTLOADER1 the default value is "1".
+#   For other partitions the default value is ${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}.
+#
+# * If both MBL_<partition>_SIZE_KiB and MBL_<partition>_SIZE_MiB are set
+#   then they must not contradict each other.
```

Comment:
What is the meaning of contradicting? They can be different?
Why we don't reduce implementation complexity and just accept the size in KiB?

Reply:
It's supposed to mean: if you specify the size in KiB and the size in MiB then
```
MBL_<partition>_SIZE_KiB == 1024 * MBL_<partition>_SIZE_MiB
```
must be true.

Yes, we could just accept the size in KiB. I was keeping `_SIZE_MiB` because:
* People will expect to be able to set the sizes in MiB because that's what Wic does.
* Most of the time, the size _must_ be a multiple of 1MiB (because that's what Wic uses).

It's just the bootloaders in non-file system partitions that are different.

If you think it would be clearer to just drop the `SIZE_MiB` variables then I'll gladly do that - it would make the code a lot cleaner.

Reply:
IDK.
I like the idea of keeping it simple, but at the same time, we will break the compatibility with the standard wic parameters.

Reply:
I've updated the error message when the size in KiB isn't 1024 times the size in MiB: https://github.com/ARMmbed/meta-mbl/pull/617/commits/a49612f990f18d056be2eff1b8a0d30180f11dce